flow-server/src/main/java/com/vaadin/flow/component/page/Page.java

```diff
@@ -655,4 +657,35 @@ public void fetchCurrentURL(SerializableConsumer<URL> callback) {
             }
         });
     }
+
+    /**
+     * Retrieves {@code document.dir} of the current UI from the browser and
+     * passes it to the {@code callback} parameter. If the {@code document.dir}
+     * has not been set explicitly, then {@code Direction.LEFT_TO_RIGHT} will be
+     * the fallback value.
+     * <p>
+     * Note that direction is fetched from the browser in an asynchronous
+     * request and passed to the callback.
+     * <p>
+     * In case you need more control over the execution you can use
+     * {@link #executeJs(String, Serializable...)} by passing
+     * {@code return document.dir}.
+     *
+     * @param callback
+     *            to be notified when the direction is resolved.
+     */
+    public void fetchPageDirection(SerializableConsumer<Direction> callback) {
+        UI.getCurrent().getPage().executeJs("return document.dir")
```

Comment:
Do we need `UI.getCurrent().getPage()` here, since the method is not static? Could it be simply a call to `executeJs`?
```suggestion
        executeJs("return document.dir")
```

Reply:
This was a left over from my draft codes, good catch!